md.py

```diff
@@ -266,17 +270,28 @@ def get_disks(self):
             disk = (self._sysfs / f"rd{d}" / "block").readlink().name
             yield f"/dev/{disk}"
 
-    def get_next_disk(self):
-        disk = (self._sysfs / "rd0" / "block").readlink().name
-
-        n = self.get_num_disks()
-        if "ram" in disk:
-            return f"/dev/ram{n}"
-        if "loop" in disk:
-            sectors = int((self._sysfs / "rd0" / "block" / "size").read_text())
-            return self._create_loop_disk(n, sectors << 9)
+    def get_next_disk(self, non_array=False):
+        if non_array:
+            dev_list = self.na_devs
+        else:
+            dev_list = self.devs
```

Comment:
Oh, this is confusing.... I guess you're adding to it here.... But it never appears to be read?

Would be much more sensible to add to the list at the end of the function with an if statement instead of twice in the middle and returning the last....

Reply:
only the length is read currently for creating new ram disks, I could have just made it a counter in the non-array case, but martin and I have been talking a bit about changing the default DISKS arg to use all DEVS without breaking journal/verify/grow tests which will use it more.

Reply:
I don't really follow that. I'm just saying it should be something like this: 

```python3
def _get_next_disk(self):
    if len(self.extra_devs):
        new_dev = self.extra_devs.pop()
    else:
        ...
    return new_dev

def get_special_disk(self):
    new_dev = self.__get_next_disk()
    self.special_devs.append(new_dev)
    return new_dev

def grow(self):
    self.wait()
    dev = self._get_next_disk(self)
    self.devs.append(dev)
    ...
```

Besides that I think the way your doing it makes sense.